### PR TITLE
Check 34 New parameter for comments

### PR DESCRIPTION
**Pull request by fidley**

### CODE REVIEW

**Comment by fidley** [comment]
I've added parameter called "Include comments?". So you can choose if in large WHEN constructs you want to count lines with comments or not. Test methods are also added.

File: src/checks/zcl_aoc_check_34.clas.testclasses.abap [hunk 1]
```diff
@@ -106,4 +108,141 @@ CLASS ltcl_test IMPLEMENTATION.
...
+      mv_lines            = lv_lines
+      mv_incl_comments    = abap_false
+      TO DATA BUFFER lv_attributes.
+
+    mo_check->put_attributes( lv_attributes ).
+
+    _code 'CASE lv_foo.'.
+    _code '  WHEN ''bar''.'.
+    _code ' ".'.
+    _code '*.'.
```
Comment (larshp) [inline src/checks/zcl_aoc_check_34.clas.testclasses.abap]: suggest replacing these with something like

```
do 10 times.
  _code '*.'.
  _code '".'.
enddo.
```

File: src/checks/zcl_aoc_check_34.clas.testclasses.abap [hunk 2]
```diff
@@ -106,4 +108,141 @@ CLASS ltcl_test IMPLEMENTATION.
...
+      mv_errty            = lv_errty
+      mv_lines            = lv_lines
+      mv_incl_comments = lv_incl_comments
+      FROM DATA BUFFER lv_attributes.                "#EC CI_USE_WANTED
+    lv_lines = 20.
+    EXPORT
+      mv_errty            = lv_errty
+      mv_lines            = lv_lines
+      mv_incl_comments    = abap_false
+      TO DATA BUFFER lv_attributes.
```
Comment (larshp) [inline src/checks/zcl_aoc_check_34.clas.testclasses.abap]: if the test class is friends with the global class, the private variable can be set directly?

File: src/checks/zcl_aoc_check_34.clas.testclasses.abap [hunk 3]
```diff
@@ -106,4 +108,141 @@ CLASS ltcl_test IMPLEMENTATION.
...
+      mv_errty            = lv_errty
+      mv_lines            = lv_lines
+      mv_incl_comments    = abap_true
+      TO DATA BUFFER lv_attributes.
+
+    mo_check->put_attributes( lv_attributes ).
+
+    _code 'CASE lv_foo.'.
+    _code '  WHEN ''bar''.'.
+    _code ' ".'.
```
Comment (larshp) [inline src/checks/zcl_aoc_check_34.clas.testclasses.abap]: as above

File: src/checks/zcl_aoc_check_34.clas.testclasses.abap (hunk 1 above)
Comment (larshp) [inline src/checks/zcl_aoc_check_34.clas.testclasses.abap]: suggest replacing these with something like

```
do 10 times.
  _code '*.'.
  _code '".'.
enddo.
```

File: src/checks/zcl_aoc_check_34.clas.testclasses.abap (hunk 2 above)
Comment (fidley) [reply src/checks/zcl_aoc_check_34.clas.testclasses.abap]: The problem is that you cannot add a friend as local test class as the definition is not known for compiler. So I think the only way to do it, is to create and global interface which will be empty, but added as a friend to each CHECK class. Then we can implement this interface in TEST class to have access to private/protected attributes and methods. 

The same way it is done in ALV grid. I can do it in this class if you want, then you can decide if to implement it in others.

**Comment by fidley** [comment]
Changes done + interface that is a Friend is added. Please check now.

**Comment by larshp** [comment]
I'm not sure I understand what `zif_aoc_check_friend` is for?

just declare them friends in the beginning of the testclass file,
```abap
class ltcl_test definition deferred.
class zcl_aoc_check_34 definition local friends ltcl_test.
```

**Comment by fidley** [comment]
image
Just to not say empty words.

**Comment by larshp** [comment]
it goes in top of the testclass include, example:
https://github.com/larshp/abapGit/blob/cbfce7a2efaa572e715c9cabd66d1ea030315cfd/src/zcl_abapgit_dot_abapgit.clas.testclasses.abap

**Comment by fidley** [comment]
OK, thanks for the hint :) Interface removed.

**Comment by larshp** [comment]
thanks 👍 

merging

**Pull request merged**